workspaces/simon-game/src/app/components/App.tsx

```diff
@@ -1,14 +1,21 @@
-import React from "react";
+import React, { useState } from "react";
 
 import { Box } from "./Box.tsx";
 import { playNote } from "../util/playNote.ts";
 import { config } from "../constants.ts";
 
 export function App() {
+  const [playerMoves, setPlayerMoves] = useState<number[]>([]);
```

Comment:
You can go a step further and mark it read-only! This way people will be forced to replace the entire array using the state setter, and they wouldn't be able to accidentally `.push` on the array.

```suggestion
  const [playerMoves, setPlayerMoves] = useState<readonly number[]>([]);
```

Reply:
This is a good tip!